File: backend/src/main/java/com/bizflow/backend/presentation/dto/response/ProductDTO.java
package com.bizflow.backend.presentation.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProductDTO {
    private Long id;
    private String name;
    private String sku;
    private BigDecimal price;
    private BigDecimal costPrice;

    // --- BẮT BUỘC PHẢI CÓ DÒNG NÀY MỚI HẾT LỖI 'cannot find symbol method stock' ---
    private Integer stock;
    // -----------------------------------------------------------------------------

    private Long categoryId;
    private String categoryName;

    private Long unitId;
    private String unitName;

    private String description;
    private String imageUrl;
    private String status;
    private Long storeId;
}

File: mobile/lib/features/home/data/models/product_dto.dart
class ProductDTO {
  final int id;
  final String name;
  final String sku;
  final int stock;
  final String? unitName;
  final String? imageUrl;
  final double price;

  ProductDTO({
    required this.id,
    required this.name,
    required this.sku,
    required this.stock,
    this.unitName,
    this.imageUrl,
    required this.price,
  });

  factory ProductDTO.fromJson(Map<String, dynamic> json) {
    return ProductDTO(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      sku: json['sku'] ?? '',
      stock: int.tryParse(json['stock'].toString()) ?? 0,
      unitName: json['unitName'],
      imageUrl: json['imageUrl'],
      price: double.tryParse(json['price'].toString()) ?? 0.0,
    );
  }
}
